### DISCUSSION TITLE
OSAL port for FreeRTOS

### DISCUSSION THREAD

**Discussion by PepeRulo** [nasa/cFS#271] [Q&A]
Hello everybody,

I’m relatively new around here but I’ve heard that there’s been a lot of discussion in the past about a community solution to porting the OSAL to FreeRTOS. Among the resources I’ve encountered, nasa/osal#620 references an archived thread from the old mailing list, but unfortunately that doesn’t seem to have a valid url anymore, at least externally. I wonder if it’s still somehow possible to access it and move it here to a discussion, and if that’d even be worth the effort, as I don’t really know what those past conversations/efforts looked like.

I do, however, know of the existence of https://github.com/METECS/OSAL-PC-FreeRTOS, which seems to be an excellent baseline, but I just started looking into it.

Has anyone heard or know of any existing community efforts for the port? 

Two famous examples on FreeRTOS mission-developed ports are NASA’s Dellingr mission and the STF-1 mission. Here is a nice resource discussing the first one:
https://ntrs.nasa.gov/citations/20170011566

**Comment by ghost** [comment]
Hi,
Currently the METECS link you mentioned is probably the best public resource. The OSAL has changed so much, that the older conversations and even the Dellingr and STF-1 software may not offer much that can be used.

What target processor and board are you interested in?
